src/lib/openapi/PropertySchema.php

```diff
@@ -82,6 +85,39 @@ public function __construct(SpecObjectInterface $property, string $name, Compone
         $this->schema = $schema;
         $this->isPk = $name === $schema->getPkName();
 
+        $onUpdate = $onDelete = null;
+        if (!empty($property->allOf[1]) &&
```

Comment:
how about foreach($property->allOf ...) { ?
This way it is not relevant in which order the foreign key or relation are defined.

Reply:
Implemented in ec2c66e2b2656a3fd9da6f228d6b03771f1a97d5